### DISCUSSION TITLE
How to get the selling_partner_id out of an access token?

### DISCUSSION THREAD

**Discussion by josei** [amzn/selling-partner-api-models#3411] [General]
During the OAuth flow, the `selling_partner_id` is sent as query param in the `redirect_uri` to the app's website, but that opens the door to the client sending any arbitrary `selling_partner_id` as long as a supplied authorization code is valid.

So, in order to authenticate the selling partner's identity, is there a way to get the `selling_partner_id` out of an access token? The Selling Partner API seems not to include some basic selling partner profile endpoint.

**Comment by MeenaAmz** [comment]
Authorization code is the link that ties the selling partner to the app. Which type of oauth flow are you using - [website](https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow) or [selling partner app store](https://developer-docs.amazon.com/sp-api/docs/selling-partner-appstore-authorization-workflow)? 

In both oauth flows Amazon already authenticates the selling partner since the selling partner has to login to Seller or vendor central to authorize the app. You could also set up additional authentication on the app side by having them register\login.

**Reply by josei** [reply]
I'm using the website oauth flow. [Step 4](https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow#step-4-your-application-exchanges-the-lwa-authorization-code-for-a-lwa-refresh-token) is where the authorization code is exchanged by the backend to get an access token.

The question is how to get the `selling_partner_id` out of the resulting access token. Isn't there an endpoint to retrieve basic access token information? I.e., retrieve which selling partner is associated with a specific access token.
